typescript_test/test.ts

```diff
@@ -1164,8 +1164,8 @@ function deepNesting() {
   const selector17: OutputSelector<
     [(state: State) => string],
     ReturnType<typeof selector16>,
-    never,
-    (s: string) => string
+    (s: string) => string,
+    never
```

Comment:
Good catch, thanks!

Reply:
I knew this would cause an issue if the other PR merged, but forgot to mention it.